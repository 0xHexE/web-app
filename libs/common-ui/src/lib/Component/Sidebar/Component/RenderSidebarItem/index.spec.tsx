import { render } from '@testing-library/react';
import { RenderSidebarItem } from './index';
import { Menu16 } from '@carbon/icons-react';

describe('RenderSidebarItem', function () {
  it('should render', function () {
    const { baseElement: baseElement1 } = render(
      <RenderSidebarItem
        item={{ key: 'test', title: 'jas', icon: <Menu16 />, kind: 'action' }}
      />
    );

    expect(baseElement1).toMatchSnapshot();

    const { baseElement: baseElement2 } = render(
      <RenderSidebarItem
        item={{
          key: 'test',
          title: 'jas',
          icon: <Menu16 />,
          kind: 'action',
          children: [
            { key: 'test', title: 'jas', icon: <Menu16 />, kind: 'action' },
          ],
        }}
      />
    );
    expect(baseElement2).toMatchSnapshot();

    const { baseElement: baseElement3 } = render(
      <RenderSidebarItem
        rootRender={true}
        item={{
          key: 'test',
          title: 'jas',
          icon: <Menu16 />,
          kind: 'action',
          children: [
            { key: 'test', title: 'jas', icon: <Menu16 />, kind: 'action' },
          ],
        }}
      />
    );
    expect(baseElement3).toMatchSnapshot();

    const { baseElement: baseElement4 } = render(
      <RenderSidebarItem
        rootRender={true}
        item={{
          key: 'test',
          kind: 'spacer',
        }}
      />
    );
    expect(baseElement4).toMatchSnapshot();

    const { baseElement: baseElement5 } = render(
      <RenderSidebarItem
        rootRender={true}
        item={{
          key: 'test',
          kind: 'spaacer' as never,
        }}
      />
    );
    expect(baseElement5).toMatchSnapshot();
  });
});
