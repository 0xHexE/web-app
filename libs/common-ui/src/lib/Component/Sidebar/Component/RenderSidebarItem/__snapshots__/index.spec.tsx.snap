// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderSidebarItem should render 1`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
</body>
`;

exports[`RenderSidebarItem should render 2`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
</body>
`;

exports[`RenderSidebarItem should render 3`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
  <div>
    <li
      class=""
    >
      <div
        aria-expanded="false"
        class=""
        role="button"
        tabindex="0"
      >
        jas
        <svg
          class=""
          size="16"
          title="Expand"
          viewBox="0 0 24 24"
        >
          <g
            class=""
          >
            <path
              clip-rule="evenodd"
              d="M6 12C6 11.4477 6.44772 11 7 11H17C17.5523 11 18 11.4477 18 12C18 12.5523 17.5523 13 17 13H7C6.44772 13 6 12.5523 6 12Z"
              fill-rule="evenodd"
            />
          </g>
          <path
            clip-rule="evenodd"
            d="M6 12C6 11.4477 6.44772 11 7 11H17C17.5523 11 18 11.4477 18 12C18 12.5523 17.5523 13 17 13H7C6.44772 13 6 12.5523 6 12Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class=""
      >
        <div
          class=""
        />
      </div>
    </li>
  </div>
</body>
`;

exports[`RenderSidebarItem should render 4`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
  <div>
    <li
      class=""
    >
      <div
        aria-expanded="false"
        class=""
        role="button"
        tabindex="0"
      >
        jas
        <svg
          class=""
          size="16"
          title="Expand"
          viewBox="0 0 24 24"
        >
          <g
            class=""
          >
            <path
              clip-rule="evenodd"
              d="M6 12C6 11.4477 6.44772 11 7 11H17C17.5523 11 18 11.4477 18 12C18 12.5523 17.5523 13 17 13H7C6.44772 13 6 12.5523 6 12Z"
              fill-rule="evenodd"
            />
          </g>
          <path
            clip-rule="evenodd"
            d="M6 12C6 11.4477 6.44772 11 7 11H17C17.5523 11 18 11.4477 18 12C18 12.5523 17.5523 13 17 13H7C6.44772 13 6 12.5523 6 12Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class=""
      >
        <div
          class=""
        />
      </div>
    </li>
  </div>
  <div>
    <span
      class=""
    />
  </div>
</body>
`;

exports[`RenderSidebarItem should render 5`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
        />
      </svg>
      <span
        class=""
      />
      jas
    </div>
  </div>
  <div>
    <li
      class=""
    >
      <div
        aria-expanded="false"
        class=""
        role="button"
        tabindex="0"
      >
        jas
        <svg
          class=""
          size="16"
          title="Expand"
          viewBox="0 0 24 24"
        >
          <g
            class=""
          >
            <path
              clip-rule="evenodd"
              d="M6 12C6 11.4477 6.44772 11 7 11H17C17.5523 11 18 11.4477 18 12C18 12.5523 17.5523 13 17 13H7C6.44772 13 6 12.5523 6 12Z"
              fill-rule="evenodd"
            />
          </g>
          <path
            clip-rule="evenodd"
            d="M6 12C6 11.4477 6.44772 11 7 11H17C17.5523 11 18 11.4477 18 12C18 12.5523 17.5523 13 17 13H7C6.44772 13 6 12.5523 6 12Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class=""
      >
        <div
          class=""
        />
      </div>
    </li>
  </div>
  <div>
    <span
      class=""
    />
  </div>
  <div />
</body>
`;
